File: web/src/mixins/switches.js
import { firestore } from "../modules/firebase"

export default {
  data() {
    return {
      switches: null,
    }
  },
  methods: {
    getSwitch(key) {
      if (!this.switchesLoaded) return false

      if (this.enableAdminSwitchOverride && this.$store.state.isAdmin) {
        return true
      } else {
        return key in this.switches && this.switches[key]
      }
    },
  },
  computed: {
    switchesLoaded() {
      return this.switches != null
    },
    enableAdminSwitchOverride() {
      return this.switches.enable_admin_switch_override
    },
    enableDraft() {
      return this.getSwitch("enable_draft")
    },
    enableFacebookLogin() {
      return this.getSwitch("enable_facebook_login")
    },
    enableInviteByEmail() {
      return this.getSwitch("enable_invite_by_email")
    },
    enableLogin() {
      return this.getSwitch("enable_login")
    },
    enablePublicLeagues() {
      return this.getSwitch("enable_public_leagues")
    },
    enableTwitterLogin() {
      return this.getSwitch("enable_twitter_login")
    },
    enablePlayerLinks() {
      return this.getSwitch("enable_player_links")
    },
    enableProjections() {
      return this.getSwitch("enable_projections")
    },
    enableRelaxedIR() {
      return this.getSwitch("enable_relaxed_ir")
    },
    enableMatchupProgress() {
      return this.getSwitch("enable_matchup_progress")
    },
    showNewRosterUITip() {
      return this.getSwitch("show_new_roster_ui_tip")
    },
    showDonationLink() {
      return this.getSwitch("show_donation_link")
    },
    donateMessage() {
      if (!this.switchesLoaded) return false
      return this.switches.donate_message
    },
    enableDiscordIntegration() {
      return this.getSwitch("enable_discord_integration")
    },
    officialDiscordLink() {
      if (!this.switchesLoaded) return false
      return this.getSwitch("official_discord_link")
    },
  },
  created() {
    this.$bind("switches", firestore.collection("public").doc("switches"))
  },
}

File: services/api/app/domain/entities/switches.py
from app.core.annotate_args import annotate_args
from app.core.base_entity import BaseEntity

# this class does not need all switches, only the ones the API cares about
# a set (replace doc) should never be done from the API, or it will override UI switches


@annotate_args
class Switches(BaseEntity):
    id = "switches"
    enable_relaxed_ir = True
    enable_game_roster_status = False
    donate_message = "Like the site?  Click here to donate and help pay the bill :)"
    enable_admin_switch_override = False
    enable_client_scoring = True
    enable_draft = True
    enable_facebook_login = False
    enable_game_roster_status = True
    enable_invite_by_email = False
    enable_login = True
    enable_matchup_progress = True
    enable_player_links = True
    enable_projections = True
    enable_public_leagues = False
    enable_relaxed_ir = True
    enable_score_testing = False
    enable_twitter_login = False
    show_donation_link = True
    show_new_roster_ui_tip = False
    enable_discord_integration = True
